src/Microsoft.DotNet.Interactive/KernelScheduler.cs

```diff
@@ -128,7 +140,7 @@ private void Run(ScheduledOperation operation)
                                     {
                                         if (!operation.TaskCompletionSource.Task.IsCompleted)
                                         {
-                                            if (t.IsCompletedSuccessfully)
+                                            if (IsCompletedSuccessfully(t))
```

Comment:
This can still be an extension method.

Reply:
I know, I will clean things up, much of this is ugly right now.

Reply:
You call this ugly? Pffff. I've seen uglier.